File: 3330-find-the-original-typed-string-i/3330-find-the-original-typed-string-i.py
class Solution:
    def possibleStringCount(self, word: str) -> int:
        n = len(word)
        possible = set()
        possible.add(word) # 原本沒出錯的情況

        i = 0
        while i < n:
            j = i
            # 找出一段連續相同的字元
            while j < n and word[j] == word[i]:
                j += 1
            length = j - i
            if length >= 2:
                 # 嘗試刪除多餘的部分，只保留一個字元
                for k in range(1, length):
                    new_word = word[:i] + word[i+k:] # 刪掉 k 個重複字元
                    possible.add(new_word)
            i = j # 繼續下一段
        
        return len(possible)

File: 3330-find-the-original-typed-string-i/3330-find-the-original-typed-string-i.cpp
class Solution {
public:
    int possibleStringCount(string word) {
        unordered_set<string> possible;
        possible.insert(word); // 原始輸入也可能是正確的

        int n = word.size();
        int i = 0;

        while(i < n){
            int j = i;
            // 找出連續相同的字元區段
            while(j < n && word[j] == word[i]){
                j++;
            }
            int len = j -i;
            if (len >= 2){
                // 將這段中的多餘字元刪除，只保留第一個
                for (int k = 1; k < len; ++k){
                    string new_word = word.substr(0, i) + word.substr(i + k);
                    possible.insert(new_word);
                }
            }
            i = j;
        }
        return possible.size();
    }
};